gapis/trace/android/trace.go

```diff
@@ -476,6 +476,22 @@ func (t *androidTracer) SetupTrace(ctx context.Context, o *service.TraceOptions)
 
 	var process tracer.Process
 	if o.Type == service.TraceType_Perfetto {
+		if pkg != nil {
+			d := pkg.Device.(adb.Device)
+			abi := pkg.ABI
+			if abi.SameAs(device.UnknownABI) {
+				abi = pkg.Device.Instance().GetConfiguration().PreferredABI(nil)
+			}
+			// For NativeBridge emulated devices opt for the native ABI of the emulator.
+			abi = d.NativeBridgeABI(ctx, abi)
+			layers := tracer.LayersFromOptions(ctx, o)
+
+			c, err := android.SetupLayers(ctx, d, pkg.Name, gapidapk.PackageName(abi), layers, true)
+			if err != nil {
+				return nil, cleanup.Invoke(ctx), log.Err(ctx, err, "Setting up the layer")
+			}
+			cleanup = cleanup.Then(c)
+		}
```

Comment:
This won't work, as we're overriding the layers again inside `perfetto/android/trace.go` (i.e. the line above you had to fix). This should be done in there instead of here.
Also, this does not ensure that `gapid.apk` is installed so the layers can be found.